functions/src/models/users/index.ts

```diff
@@ -0,0 +1,147 @@
+import { FirestoreDataConverter } from '@google-cloud/firestore';
+import { IsInt, IsNotEmpty, IsNumber, IsObject, IsString, Max, Min } from 'class-validator';
+import { firestore } from 'firebase-admin';
+
+// eslint-disable-next-line import/no-cycle
+import { IQuestionnaire } from '../questionnaires';
+import DocumentData = firestore.DocumentData;
+import DocumentReference = firestore.DocumentReference;
+
+export interface IUser extends DocumentData {
+  cavQuestionnaireRef: DocumentReference<IQuestionnaire> | null;
+  pinQuestionnaireRef: DocumentReference<IQuestionnaire> | null;
+  averageRating: number;
+  casesCompleted: number;
+  requestsMade: number;
+  username: string;
+  displayName: string | null;
+}
```

Comment:
I just realised that we currently have no way of linking firebase auth objects with IUser objects right? We need some way of linking "Firebase ID tokens" to "IUser" objects, correct?

So we probably need to add a uid field to IUser, right? (in the future we probably want it to be optional, to represent people that haven't logged-in, but that can be later). Perhaps making this a `string[]` in case we need to attach multiple auth objects to a single person?

Reply:
ahh... adding this to the `User` object then makes it public data, and I imagine we don't want to share people's tokens with one another (even if they can't be spoofed).

Perhaps the IUser object needs to be private by default, and have a public sub-collection? something like "PublicProfile"?

Reply:
There’s an easier way to do this :)